packages/models-library/tests/test_errors_classes.py

```diff
@@ -0,0 +1,118 @@
+# pylint: disable=protected-access
+# pylint: disable=redefined-outer-name
+# pylint: disable=unused-argument
+# pylint: disable=unused-variable
+# pylint: disable=no-member
+
+
+from datetime import datetime
+from typing import Any
+
+import pytest
+from models_library.errors_classes import OsparcErrorMixin
+
+
+def test_get_full_class_name():
+    class A(OsparcErrorMixin):
+        ...
+
+    class B1(A):
+        ...
+
+    class B2(A):
+        ...
+
+    class C(B2):
+        ...
+
+    class B12(B1, ValueError):
+        ...
+
+    assert B1._get_full_class_name() == "A.B1"
+    assert C._get_full_class_name() == "A.B2.C"
+    assert A._get_full_class_name() == "A"
+
+    # diamond inheritance (not usual but supported)
+    assert B12._get_full_class_name() == "ValueError.A.B1.B12"
```

Comment:
Could you please add a note/docstring to the `_get_full_class_name` that points out the last term of the chain `X.Y.Z.ActualErrorClassName` is the name of the class?

Reply:
@GitHK Isn't that what you would expect?  In any case, the idea here is not to use this method but that it auto-generates the `error.code` . Look at `__new__` method and the examples